server/odc-service/src/main/java/com/oceanbase/odc/service/db/session/DefaultDBSessionManage.java

```diff
@@ -198,33 +210,182 @@ private List<KillSessionResult> doKillSessionOrQuery(
+        boolean isEnabledGlobalClientSession =
+                checkGlobalClientSessionEnabled(connectionSession, obProxyVersion, obVersion);
+        boolean isSupportedOracleModeKillSession = checkOracleModeKillSessionSupported(obVersion, connectionSession);
+        List<JdbcGeneralResult> finalResults = new ArrayList<>();
+        for (JdbcGeneralResult jdbcGeneralResult : results) {
             try {
                 jdbcGeneralResult.getQueryResult();
             } catch (Exception e) {
-                if (StringUtils.contains(e.getMessage(), "Unknown thread id")) {
-                    try {
-                        log.info("Kill query/session Unknown thread id error, try direct connect observer");
-                        directLinkServerAndExecute(sqlTuple.getExecutedSql(),
-                                connectionSession);
-                        return JdbcGeneralResult.successResult(sqlTuple);
-                    } catch (Exception e1) {
-                        log.warn("Failed to direct connect observer {}", e1.getMessage());
-                    }
+                if (isUnknownThreadIdError(e)) {
+                    jdbcGeneralResult = handleUnknownThreadIdError(connectionSession,
+                            jdbcGeneralResult, isDirectedOBServer,
+                            isEnabledGlobalClientSession, isSupportedOracleModeKillSession);
                 } else {
-                    log.warn("Failed to execute sql in kill session scenario, sqlTuple={}", sqlTuple, e);
+                    log.warn("Failed to execute sql in kill session scenario, sqlTuple={}",
+                            jdbcGeneralResult.getSqlTuple(), e);
                 }
             }
+            finalResults.add(jdbcGeneralResult);
+        }
+        return finalResults;
+    }
+
+    private static boolean checkObServerDirected(ConnectionSession connectionSession) {
+        String sql;
+        if (OB_MYSQL == connectionSession.getDialectType()) {
+
+            sql = "select PROXY_SESSID from oceanbase.gv$ob_processlist where ID =(select id from oceanbase.gv$ob_processlist where info like '%gv$ob_processlist%');";
+        } else {
+            sql = "select PROXY_SESSID from gv$ob_processlist where ID =(select id from gv$ob_processlist where info like '%gv$ob_processlist%');";
+        }
+        return connectionSession.getSyncJdbcExecutor(ConnectionSessionConstants.BACKEND_DS_KEY)
+                .queryForObject(sql, String.class) == null;
+    }
+
+    private String getObProxyVersion(ConnectionSession connectionSession, boolean isDirectedOBServer) {
+        if (isDirectedOBServer) {
+            return null;
+        }
+        try {
+            return connectionSession.getSyncJdbcExecutor(ConnectionSessionConstants.BACKEND_DS_KEY)
+                    .queryForObject("select proxy_version()", String.class);
+        } catch (Exception e) {
+            log.warn("Failed to obtain the OBProxy version number: {}", e.getMessage());
```

Comment:
log.info , here failed is expected for old version obproxy

Reply:
ok